reya_trading/auth/signatures.py

```diff
@@ -0,0 +1,332 @@
+"""
+Signature generation utilities for Reya Trading API authentication.
+
+This module provides tools for creating EIP-712 signatures for order creation
+and message signatures for order cancellation.
+"""
+import time
+import json
+import secrets
+from typing import Dict, Any, List, Union, Tuple, Optional
+from decimal import Decimal
+from eth_abi import encode
+from eth_account.messages import encode_typed_data
+
+from eth_account import Account
+from eth_account.messages import encode_defunct
+from web3 import Web3
+
+from ..config import TradingConfig
+from ..constants.enums import ConditionalOrderType, ConditionalOrderStatus
+
+
+class SignatureGenerator:
+    """Generate signatures for Reya Trading API requests."""
+
+    def __init__(self, config: TradingConfig):
+        """
+        Initialize the signature generator with configuration.
+        
+        Args:
+            config: Trading API configuration
+        """
+        self.config = config
+        self._private_key = config.private_key
+        self._chain_id = config.chain_id
+        
+        if not self._private_key:
+            raise ValueError("Private key is required for signing")
+        
+        self._orders_gateway_address = (
+            config.orders_gateway_address or 
+            config.default_orders_gateway_address
+        )
+        
+        # Conditional orders use the same address for now
+        self._conditional_orders_address = config.default_conditional_orders_address
```

Comment:
hmm, there's no smart contract called conditional orders on-chain, we just have the orders gateway contract that manages executions

Reply:
we should use orders gateway address everywhere in place of _conditional_orders_address

Reply:
Fixed it now !